### ISSUE TITLE
Invalid inline ASM when compiling for ARM

### ISSUE THREAD

**Issue by omarroth** [crystal-lang/crystal#7039]
Unfortunately, it looks like #6897 introduced an issue in how asm is generated for arm platforms. Cross-compiling produces:
```
<inline asm>:8:20: error: invalid operand for instruction
        ldr    sp, r1
                   ^
LLVM ERROR: Error parsing inline asm
```

The issue appears to be caused by [this](https://github.com/crystal-lang/crystal/blob/master/src/fiber/arm.cr#L34) line, which appears to be a typo.
Changing
```
ldr    sp, $1
```
to 
```
ldr    sp, [$1]
```
Produces a valid binary. Unfortunately running it on an arm device (Raspberry Pi B+ v1.2) produces:
```
Invalid memory access (signal 11) at address 0x0
[0xeb8e7c] ???
[0x22ceec] __crystal_sigfault_handler +212
```

Can reproduce with:
```
$ crystal build samples/tree.cr --cross-compile --target arm-unknown-linux-gnueabihf
```

On host:
```
$ uname -srvmpio
Linux 4.18.16-arch1-1-ARCH #1 SMP PREEMPT Sat Oct 20 22:06:45 UTC 2018 x86_64 unknown unknown GNU/Linux

$ crystal -v
Crystal 0.27.0 (2018-11-02)

LLVM: 6.0.1
Default target: x86_64-pc-linux-gnu
```

On raspberry pi:
```
$ uname -srvmpio
Linux 4.14.71+ #1145 Fri Sep 21 15:06:38 BST 2018 armv6l unknown unknown GNU/Linux

$ crystal -v
Crystal 0.27.0 (2018-11-05)

LLVM: 3.8.1
Default target: armv6--linux-gnueabihf
```

The Pi compiler was cross-compiled using steps mentioned [here](https://stackoverflow.com/a/42796648). The compiler is able to format, print help messages, and other simple tasks, but attempting to build some of the samples produces the above mentioned:
```
crystal build samples/tree.cr 
Invalid memory access (signal 11) at address 0x0
[0xeb8e7c] ???
[0x22ceec] __crystal_sigfault_handler +212
```

**Comment by ysbaddaden** [comment]
The calling form was changed, we no longer need to dereference a pointer, but I think I improperly changed the ASM for ARM. The correct ASM should be `mov sp, $1` on lines 34 and line 48 of `src/fiber/arm.cr` (as for AArch64). Can you try?

**Comment by omarroth** [comment]
Works great, and thanks for the quick response! Mind if I open a PR?

**Comment by RX14** [comment]
@omarroth please do

**Comment by omarroth** [comment]
Fixed with #7041.
